File: app/src/main/java/fr/quintipio/simplyPassword/controller/FolderAdapter.java
package fr.quintipio.simplyPassword.controller;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import fr.quintipio.simplyPassword.R;
import fr.quintipio.simplyPassword.model.Dossier;

import java.util.List;

public class FolderAdapter extends ArrayAdapter<Dossier> {

    public FolderAdapter(Context context, List<Dossier> dossiers) {
        super(context, 0, dossiers);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        if(convertView == null){
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.row_folder,parent, false);
        }

        FolderViewHolder viewHolder = (FolderViewHolder) convertView.getTag();
        if(viewHolder == null){
            viewHolder = new FolderViewHolder();
            viewHolder.nomDossier = convertView.findViewById(R.id.nomDossier);
            convertView.setTag(viewHolder);
        }

        Dossier dossier = getItem(position);

        viewHolder.nomDossier.setText(dossier.getTitre());

        return convertView;
    }

    private class FolderViewHolder{
        public TextView nomDossier;
    }
}

File: app/src/main/java/fr/quintipio/simplyPassword/controller/FolderAdapter.kt
package fr.quintipio.simplyPassword.controller

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import fr.quintipio.simplyPassword.R
import fr.quintipio.simplyPassword.model.Dossier

class FolderAdapter(context: Context, dossiers: MutableList<Dossier>) : ArrayAdapter<Dossier>(context, 0, dossiers) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView

        if (convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.row_folder, parent, false)
        }

        var tag = convertView?.tag
        if(tag == null) {
            tag = FolderViewHolder()
            tag.nomDossier = convertView?.findViewById(R.id.nomDossier)
            convertView?.tag = tag
        }
        val dossier = getItem(position)
        tag = tag as FolderViewHolder
        tag.nomDossier!!.text = dossier!!.titre
        return convertView!!
    }

    private inner class FolderViewHolder {
        var nomDossier: TextView? = null
    }
}
